frappe/core/doctype/system_settings/system_settings.js

```diff
@@ -38,5 +42,8 @@ frappe.ui.form.on("System Settings", {
 			// Clear cache after saving to refresh the values of boot.
 			frappe.ui.toolbar.clear_cache();
 		}
-	}
+	},
+	week_starts_on(frm) {
+		frm.re_setup_moment = true;
```

Comment:
why not just setup moment when `week_starts_on` is changed here? No need to put it in refresh?

Reply:
`setup_moment` uses the value from `frappe.sys_default`. 

`frappe.sys_default` is updated inside refresh function hence setting up moment in `refresh` to get the latest data after the change.